### ISSUE TITLE
Support redirectToDefaultLang in static mode?

### ISSUE THREAD

**Issue by justincy** [vinissimus/next-translate#77]
It seems like we could support `redirectToDefaultLang` in static mode by using a [catch-all route](https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes) at the top level. That page could detect the language and perform the redirect accordingly.

**Comment by justincy** [comment]
Here's an example of what I mean, as borrowed from the README:

```
├── ca
│   ├── about.js
│   ├── index.js
│   └── nested
│       └── index.js
├── en
│   ├── about.js
│   ├── index.js
│   └── nested
│       └── index.js
├── es
│   ├── about.js
│   ├── index.js
│   └── nested
│       └── index.js
└── [...path].js
```

**Comment by aralroca** [comment]
@justincy thank you a lot for your proposal!! 🚀

With "redirects" in static mode, do you mean that the browser does these redirects, doesn't it?

I guess that it will be interesting to add the new config prop `redirectToDefaultLang` for both (static mode and for a custom server middleware). 


*  **`redirectToDefaultLang=false`** as default, for backwards compatibility.
* **`redirectToDefaultLang=true`** (if `en` is the defaultLanguage)-> 
    * **Static mode**: It redirects on the browser level, routes as `/some/route` to `/en/some/route`. 
    * **Custom server middleware**: It redirects with a status 301 on the server, routes as `/some/route` to `/en/some/route`.

I will investigate the catch-all route, I never used before! 😄

**Comment by aralroca** [comment]
I remember than @BjoernRave did a similar proposal here: https://github.com/vinissimus/next-translate/issues/51, and we decided at that moment to don't bring this responsibility to the lib. However, if that feature is commonly used and it doesn't increase the size of the lib, we can rethink it in order to implement it.

**Comment by aralroca** [comment]
Maybe we can provide more than one redirect mode:

* `redirectMode`:  
   * **`'none'`** (by default) - No redirects
   * **`'browserLang'`**  - Redirect to the lang of the browser when the language is not present on the URL. If the browser language is not a provided language, we redirect to `defaultLanguage`.
   * **`'defaultLang'`** - Redirect directly to the `defaultLanguage` when the language is not present on the URL.

**Comment by justincy** [comment]
> Maybe we can provide more than one redirect mode:
> 
> * `redirectMode`:
>   
>   * **`'none'`** (by default) - No redirects
>   * **`'browserLang'`**  - Redirect to the lang of the browser when the language is not present on the URL. If the browser language is not a provided language, we redirect to `defaultLanguage`.
>   * **`'defaultLang'`** - Redirect directly to the `defaultLanguage` when the language is not present on the URL.

That's a good idea. But if you don't want to bother with language detection, you could just leave the current functionality of redirecting to the default lang and add language detection later.

**Comment by aralroca** [comment]
Great!! 🎉 In order to finish this issue:

* [ ] Add docs on README explain it how it works
* [ ] Implement `redirectToDefaultLang` for no static mode (custom server with a middleware)
* [ ] Add tests if is possible

**Comment by aralroca** [comment]
@justincy I did a prerelease under `0.11.0-canary.1 ` with the support of `redirectToDefaultLang` in static mode.

**Comment by aralroca** [comment]
@justincy I realized that it works using some path, but when the path is just the homepage (localhost:3000), it displays a 404:

![image](https://user-images.githubusercontent.com/13313058/76652922-dd79b800-6567-11ea-8cd3-079b05e4a45d.png)

**Comment by justincy** [comment]
Thanks. I'll look into that bug.

**Comment by aralroca** [comment]
@justincy I did a PR fixing some issues that I found (they are described in the PR). If you can take a look!

**Comment by justincy** [comment]
> Great!! 🎉 In order to finish this issue:
> 
> * [ ]  Add docs on README explain it how it works
> * [ ]  Implement `redirectToDefaultLang` for no static mode (custom server with a middleware)
> * [ ]  Add tests if is possible

Updating the README and adding tests makes sense to me. But the bit about custom server and middleware doesn't make sense to me. Doesn't it already exist [here](https://github.com/vinissimus/next-translate/blob/89e203863e3d601e8fe26bc79908a9afac7f15b7/src/i18nMiddleware.js#L32)? Or is there something else you're asking for?

**Comment by aralroca** [comment]
Wow @justincy you are right! This is already existing!! hah! Ok, perfect!

**Comment by aralroca** [comment]
This will be available on `0.11.0` release
